### PR TITLE
TRAN-26068 Implements the sensitive data stream

### PR DESCRIPTION
**Pull request by GillesRasigade**
Remove sensitive data from the logs.

screenshot from 2018-09-25 12-34-16

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -148,6 +154,27 @@ describe('index.js', () => {
...
+          req: {
+            token: 'My personal token'
+          }
+        }, 'Logging amazingly sensitive data!');
+        const message = logs.shift();
+
+        expect(message.password).to.equal('__SENSITIVE_DATA__');
+        expect(message.headers['accept-language']).to.equal('fr-FR');
+        expect(message.headers.authorization).to.equal('__SENSITIVE_DATA__');
+        expect(message.req.token).to.equal('__SENSITIVE_DATA__');
```
Comment (abdelhakimbendjabeur) [inline test/index.test.js]: Just asking why haven't you considered a `deepEqual` here, are there other fields you can't control?

File: index.js [hunk 2]
```diff
@@ -21,6 +23,11 @@ if (process.env.USE_BUNYAN_PRETTY_STREAM === 'true') {
   const prettyStdOut = new PrettyStream();
   prettyStdOut.pipe(process.stdout);
   config.streams.push({ type: 'raw', level: loggerLevel, stream: prettyStdOut });
+} else if (process.env.LOGGER_USE_SENSITIVE_DATA_STREAM !== 'false') {
```
Comment (SMontenez) [inline index.js]: Does this mean that if `process.env.USE_BUNYAN_PRETTY_STREAM` is set to `true` then sensitive data won't be filtered at all ?

File: test/index.test.js (hunk 1 above)
Comment (GillesRasigade) [reply test/index.test.js]: Yes because the final result is controlled by `lodash`.

File: index.js (hunk 2 above)
Comment (GillesRasigade) [reply index.js]: Yes

**Pull request merged**